spirv_glsl.cpp

```diff
@@ -2437,22 +2448,35 @@ string CompilerGLSL::legacy_tex_op(const std::string &op, const SPIRType &imgtyp
 		break;
 	}
 
+	bool avoid_lod = check_lod_allowed(lod);
```

Comment:
Probably clearer to check something positive like `bool use_explicit_lod = check_explicit_lod_allowed(lod)` or something like that.